### PR TITLE
node: Add TOML configuration

### LINKED ISSUE #831: node: Add TOML configuration support
#### Summary
Add the possibility to specify the configuration using a TOML file

#### Additional context
N/A

### PR DESCRIPTION
**Pull request by herr-seppia**
Resolves #831

### CODE REVIEW

File: node/src/bin/main.rs [hunk 1]
```diff
@@ -10,9 +10,63 @@ use node::mempool::MempoolSrv;
...
+        "plain" => {
+            let subscriber = subscriber.with_ansi(false).finish();
+            tracing::subscriber::set_global_default(subscriber)?;
+        }
+        "coloured" => {
+            let subscriber = subscriber.finish();
+            tracing::subscriber::set_global_default(subscriber)?;
+        }
+        _ => unreachable!(),
+    };
```
Comment (goshawk-3) [inline node/src/bin/main.rs:69]: Could you move this in a separate function to make `main` more readable?

File: node/src/data/ledger.rs [hunk 2]
```diff
@@ -15,7 +15,7 @@ pub struct Transaction {
 
 impl Transaction {
     pub fn hash(&self) -> [u8; 32] {
-        self.inner.hash().to_bytes().into()
+        self.inner.hash().to_bytes()
```
Comment (goshawk-3) [inline node/src/data/ledger.rs:18]: Ledger.rs is deleted in `node-data`-related PR and this clippy warning was addressed too.

Comment (herr-seppia) [reply node/src/data/ledger.rs:18]: You're right, I'm going to rebase this PR on top of your changes

**Review by goshawk-3 — APPROVED** [review]
LGTM

**Pull request merged**